src/app/pages/datasets/components/dataset-management/dataset-management.component.ts

```diff
@@ -20,7 +22,20 @@ import { WebSocketService } from 'app/services';
 })
 export class DatasetsManagementComponent implements OnInit {
   isLoading$ = this.datasetStore.isLoading$;
-  selectedDataset$ = this.datasetStore.selectedDataset$;
+  isLoadingDataset = false;
+  selectedDataset$ = this.datasetStore.selectedDataset$.pipe(
+    filter((dataset) => !!dataset?.id),
+    tap(() => {
+      this.isLoadingDataset = true;
+      this.cdr.markForCheck();
+    }),
+    switchMap((dataset) => this.ws.call('pool.dataset.query', [[['id', '=', dataset.id]]])),
+    map((datasets) => datasets[0]),
+    tap(() => {
+      this.isLoadingDataset = false;
+      this.cdr.markForCheck();
+    }),
+  );
   selectedParentDataset$ = this.datasetStore.selectedParentDataset$;
```

Comment:
This line is unused, but it was used before.

Reply:
Yes, I'll remove it. Before `selectedParentDataset` was passed from `Dataset Management` to `Dataset Details Panel` and then to `ZFS Encryption Card`. Now it is loaded in `Dataset Details Panel` and then passed to `ZFS Encryption Card`